sktime/datasets/_single_problem_loaders.py

```diff
@@ -1438,3 +1439,233 @@ def load_forecastingdata(
     return load_tsf_to_dataframe(
         path_to_file, replace_missing_vals, value_column_name, return_type
     )
+
+
+def load_m5(
+    extract_path=None,
+    folder_path=None,
+    include_events=False,
+    test=False,
+):
+    """Fetch M5 dataset from https://zenodo.org/records/12636070 .
+
+    Downloads and extracts dataset if not already downloaded. Fetched dataset is
+    in the standard .csv format and loaded into an sktime-compatible in-memory
+    format (`pd_multiindex_hier`).
+
+    Parameters
+    ----------
+    extract_path : str, optional (default=None)
+        The path to extract the data to. If the provided path doesn't
```

Comment:
* I would be more specific here on what it means to "already contain the data" - a path with all files present, from a previous run of the function?
* be more specific about the kind of string expected here. If relative, what is it going to be relative to?

Reply:
```
extract_path : str, optional (default=None)

If provided, the path should use the appropriate path separators for the operating system.(e.g., forward slashes '/' for Unix-based systems, backslashes '\\' for Windows).
- If `extract_path` is provided:
    - Check if the required files (.csv) are present at the given `extract_path`.
    - If files are not found, check if the directory "m5-forecasting-accuracy" exists within the `extract_path`. Useful when the function has already run previously with the same path.
    - If the directory does not exist, download and extract the data  into "m5-forecasting-accuracy" folder in the `extract_path`.
    - If the directory exists, takes the path to the existing directory.
- if `extract_path` is None:
    - Check if the directory "m5-forecasting-accuracy" exists within the module level.
    - If the directory exists, takes path to current directory. Useful when the function has already run previously without any path.
    - If the directory does not exist, download and extract the data into "m5-forecasting-accuracy" folder at the module level.
```

Please let me know if i have to add anything or missed something!!